scripts/lib/api/htmlToMd.ts

```diff
@@ -102,14 +102,14 @@ function prepareHandlers(meta: Metadata): Record<string, Handle> {
         return buildMathExpression(node, "inlineMath");
       }
 
-      if (node.properties.id && node.properties.className?.includes("target")) {
+      if (
+        node.properties.id &&
+        (node.properties.className?.includes("target") ||
+          node.children.length === 0)
+      ) {
         return [buildSpanId(node.properties.id), ...all(h, node)];
       }
 
-      if (node.properties.id && node.children.length === 0) {
-        return buildSpanId(node.properties.id);
-      }
-
```

Comment:
The difference between these two ifs was the call to all(h, node). This merge should be safe given that the `all` method transforms the children of a hast parent to mdast or returns an empty array if no children are found.